src/ledger/backend/readconn.ts

```diff
@@ -48,4 +56,25 @@ export class LedgerReadConn implements LedgerReader {
 	erdstall(): Address {
 		return Address.fromString(this.contract.address);
 	}
+
+	async getNftMetadata(token: Address, id: bigint): Promise<NFTMetadata> {
+		const tokenS = token.toString();
+		if (this.metadataUriCache.has(tokenS)) {
+			return this.metadataUriCache.get(tokenS)!;
+		}
+
+		// TODO: Create IMetadata contract interface for which bindings can be
+		// generated, so we have a single interface over which we can query
+		// metadata for different contract implementations in the future.
```

Comment:
What other token standards, besides ERC721, have the exact same public function `tokenURI(uint256)`?

Reply:
Let's think about generalization when we start handling other token types. E.g., in the case of 1155, has it the same interface with same parameters?

Reply:
Most likely not. I just thought of some exotic Tokens which might be used in the wild but differ in the way they report their `tokenURI()`. But you are right, let's talk about it when it really matters again ^^